pkg/name/repository_test.go

```diff
@@ -120,3 +120,18 @@ func TestRepositoryBadDefaulting(t *testing.T) {
 		t.Errorf("IsBadErrName == false: %v", err)
 	}
 }
+
+func TestRepositoryChildren(t *testing.T) {
+	repo, err := NewRepository("ubuntu", Insecure)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tag := repo.Tag("foo")
+	if got, want := tag.Scheme(), "http"; got != want {
```

Comment:
Why not compare the whole `tag.String()`?

Reply:
I mostly wanted to make sure options were preserved, but added a String() check too.